streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -346,20 +346,24 @@ private void runLoop() {
             // try to process one fetch record from each task via the topology, and also trigger punctuate
             // functions if necessary, which may result in more records going through the topology in this loop
             if (!activeTasks.isEmpty()) {
+                long totalProcessTime = 0L;
+
                 for (StreamTask task : activeTasks.values()) {
                     long startProcess = time.milliseconds();
 
                     totalNumBuffered += task.process();
                     requiresPoll = requiresPoll || task.requiresPoll();
 
-                    sensors.processTimeSensor.record(time.milliseconds() - startProcess);
+                    totalProcessTime += (time.milliseconds() - startProcess);
 
                     maybePunctuate(task);
 
                     if (task.commitNeeded())
                         commitOne(task, time.milliseconds());
                 }
 
+                sensors.processTimeSensor.record(totalProcessTime);
```

Comment:
This is actually recording the total process time for more than one record right?